File: src/main/java/com/github/sorusclient/client/adapter/v1_8_9/EntityImpl.java
package com.github.sorusclient.client.adapter.v1_8_9;

import com.github.sorusclient.client.adapter.IEntity;
import v1_8_9.net.minecraft.entity.Entity;

public class EntityImpl implements IEntity {

    protected final Entity entity;

    public EntityImpl(Entity entity) {
        this.entity = entity;
    }

    @Override
    public double getX() {
        return this.entity.x;
    }

    @Override
    public double getY() {
        return this.entity.y;
    }

    @Override
    public double getZ() {
        return this.entity.z;
    }

}

File: version/1.18.2/src/main/kotlin/com/github/sorusclient/client/adapter/v1_18_2/EntityImpl.kt
package com.github.sorusclient.client.adapter.v1_18_2

import com.github.sorusclient.client.adapter.IEntity
import v1_18_2.net.minecraft.entity.Entity

open class EntityImpl(protected val entity: Entity) : IEntity {

    override val x: Double
        get() = entity.x

    override val y: Double
        get() = entity.y

    override val z: Double
        get() = entity.z

}